common/src/primitives/per_thousand.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use serialization::{Decode, Encode, Error, Input, Output};
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
+pub struct PerThousand(u16);
+
+impl PerThousand {
+    pub fn new(value: u16) -> Option<Self> {
+        if value <= 1000 {
+            Some(Self(value))
+        } else {
+            None
+        }
+    }
+
+    pub fn value(&self) -> u16 {
+        self.0
+    }
+}
+
+impl Encode for PerThousand {
+    fn encode_to<T: Output + ?Sized>(&self, dest: &mut T) {
+        dest.write(&self.0.to_be_bytes());
```

Comment:
Why `to_be_bytes` is used? I think for consistency we should use default (little-endian) encoding (`self.0.encode_to(dest)`). Decoding would be a bit easier too:
```
Self::new(u16::decode(input)?).ok_or(serialization::Error::from(
            "PerThousand deserialization failed",
        ))
```

Reply:
I actually agree. In fact, I don't see a need to implement Encode at all manually.